src/main/java/com/basho/riak/client/core/query/search/YokozunaIndex.java

```diff
@@ -24,7 +24,8 @@
 {
     private final String name;
     private final String schema;
-    
+    private volatile Integer nVal;
```

Comment:
Yep. Since we don't have any synchronizing `.build()` function, the volatile will force a synchronization of an update across cached copies of the variable in other threads. Atomic vars would be overkill here.